src/frontend/src/handler/create_table.rs

```diff
@@ -1349,6 +1349,29 @@ pub async fn handle_create_table(
 ) -> Result<RwPgResponse> {
     let session = handler_args.session.clone();
 
+    // CDC connector ban check
+    if let Some(connector) = handler_args.with_options.get_connector() {
```

Comment:
Also, how about implementing a function in the `WithPropertiesExt` to check for the non shared source cdc connector usage? https://github.com/risingwavelabs/risingwave/blob/b6524f9d6ad9156d54afb47194bfd7d59353e153/src/connector/src/with_options.rs#L152